test/CMakeLists.txt

```diff
@@ -4,6 +4,19 @@ set_target_properties(gtest_main PROPERTIES FOLDER "Dependencies")
 set_target_properties(gmock PROPERTIES FOLDER "Dependencies")
 set_target_properties(gmock_main PROPERTIES FOLDER "Dependencies")
 
+# common things that the tests need
```

Comment:
this was a nice little clean up. i took the 3 test utility headers we had and combined them into one header with an implementation file. i did the same with gurka. then i made a new test library out of them with this target which is really convenient because then all the test targets can just link this. cmake automatically handles giving them visibility of include directories and other dependent libraries (ie libvalhalla and the gtest libs). so now the unit test targets (and the bench targets) are much simplier. i measured a 6% speed up of a full build from no ccache using a single thread. margins will be very low for high core systems but our CI only has 2 cores so hopefuly this will speed up CI a tiny bit